airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -212,52 +165,70 @@ async def test_read_sequential_slices(
             pytest.skip("Skipping new incremental test based on acceptance-test-config.yml")
             return
 
-        threshold_days = getattr(inputs, "threshold_days") or 0
-        stream_mapping = {stream.stream.name: stream for stream in configured_catalog_for_incremental.streams}
+        output_1 = await docker_runner.call_read(connector_config, configured_catalog_for_incremental)
+        records_1 = filter_output(output_1, type_=Type.RECORD)
+        states_1 = filter_output(output_1, type_=Type.STATE)
 
-        output = await docker_runner.call_read(connector_config, configured_catalog_for_incremental)
-        records_1 = filter_output(output, type_=Type.RECORD)
-        states_1 = filter_output(output, type_=Type.STATE)
+        # We sometimes have duplicate identical state messages in a stream which we can filter out to speed things up
+        unique_state_messages = [message for index, message in enumerate(states_1) if message not in states_1[:index]]
 
-        assert states_1, "Should produce at least one state"
-        assert records_1, "Should produce at least one record"
+        # Important!
+
+        # There is only a small subset of assertions we can make
+        # in the absense of enforcing that all connectors return 3 or more state messages
+        # during the first read.
+
+        # To learn more: https://github.com/airbytehq/airbyte/issues/29926
+        if len(unique_state_messages) < 3:
+            pytest.skip("Skipping test because there are not enough state messages to test with")
+            return
+
+        assert records_1, "First Read should produce at least one record"
 
         # For legacy state format, the final state message contains the final state of all streams. For per-stream state format,
         # the complete final state of streams must be assembled by going through all prior state messages received
         is_per_stream = is_per_stream_state(states_1[-1])
-        if is_per_stream:
-            latest_state = construct_latest_state_from_messages(states_1)
-        else:
-            latest_state = states_1[-1].state.data
 
-        for record_value, state_value, stream_name in records_with_state(records_1, latest_state, stream_mapping, cursor_paths):
-            assert (
-                record_value <= state_value
-            ), f"First incremental sync should produce records younger or equal to cursor value from the state. Stream: {stream_name}"
+        # To avoid spamming APIs we only test a fraction of batches (10%) and enforce a minimum of 10 tested
+        min_batches_to_test = 5
+        sample_rate = len(unique_state_messages) // min_batches_to_test
 
-        checkpoint_messages = filter_output(output, type_=Type.STATE)
+        mutating_stream_name_to_per_stream_state = dict()
+        for idx, state_message in enumerate(unique_state_messages):
+            assert state_message.type == Type.STATE
```

Comment:
This assertion is probably redundant as `unique_state_message` is built from a filtered output of `Type.STATE`.